pkg/transfer/local/pull_test.go

```diff
@@ -0,0 +1,189 @@
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              false,
+			ExpectedPlatform:   transfer.UnpackConfiguration{},
+		},
+		{
+			Name: "Match linux/amd64 and native snapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/amd64"),
+				Snapshotter: "native",
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/amd64"),
+				Snapshotter: "native",
+			},
+		},
+		{
+			Name: "Match linux/arm64 and native snapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm64"),
+				Snapshotter: "native",
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm64"),
+				Snapshotter: "native",
+			},
+		},
+		{
+			Name: "Match linux/arm and defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+		{
+			Name: "Platform linux/amd64 input only match with defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform: platforms.MustParse("linux/amd64"),
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/amd64"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+		{
+			Name: "Platform linux/arm64 input only match with defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform: platforms.MustParse("linux/arm64"),
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm64"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+		{
+			Name: "Platform linux/arm input only match with defaultSnapshotter",
+			UnpackConfig: transfer.UnpackConfiguration{
+				Platform: platforms.MustParse("linux/arm"),
+			},
+			SupportedPlatforms: supportedPlatforms,
+			Match:              true,
+			ExpectedPlatform: transfer.UnpackConfiguration{
+				Platform:    platforms.MustParse("linux/arm"),
+				Snapshotter: containerd.DefaultSnapshotter,
+			},
+		},
+	} {
+
+		t.Run(testCase.Name, func(t *testing.T) {
+			m, sp := getSupportedPlatform(testCase.UnpackConfig, testCase.SupportedPlatforms)
+			if m == testCase.Match {
+				if sp.SnapshotterKey == testCase.ExpectedPlatform.Snapshotter {
```

Comment:
I think you should check `sp` should be empty if the match is false, IIUC.

Reply:
Added the check.